pkgs/top-level/ocaml-packages.nix

```diff
@@ -357,7 +357,12 @@ let
       then pkgs.dune_2
       else throw "dune_2 is not available for OCaml ${ocaml.version}";
 
-    dune_3 = callPackage ../development/tools/ocaml/dune/3.nix { };
+    dune_3 =
+      if lib.versionAtLeast ocaml.version "4.08"
```

Comment:
This seems correct (like dune_2 upper) but do you know where the version range is mentioned. IIRC dune require latest ocaml - 2 majors version for building. I can't find in the doc mention about any version.

Reply:
e.g. https://github.com/ocaml/opam-repository/blob/d9d1fc2cde92926573cad64853c29d02dbd5e2a6/packages/dune/dune.3.6.0/opam#L45

Reply:
Ok for this one thanks. For the bound 4.02 I didn't find it on ocamlfind-secondary-compiler I just found this line in README https://github.com/ocaml/dune#requirements

I had this in mind https://dune.build/blog/dune-2-coming-soon/

> For this reason, starting from version 2.0.0 Dune will require a recent version of OCaml to build itself. Dune 2.0.0 will require 4.08.0, when 4.09.0 is released we will immediately start using its new features and so on. But more importantly, we will immediatly start using multicore when it becomes available, since Dune is a project that is likely to benefit from it.